nixos/modules/services/printing/lprint.nix

```diff
@@ -0,0 +1,520 @@
+      group = "lprint";
+      extraGroups = [ "usb" "lp" ];
+    };
+
+    etc."lprint.conf".text = ''
+      ${lib.optionalString (lprintConfig.papplOptions.dnssdName != null) "DNSSDName ${lprintConfig.papplOptions.dnssdName}"}
+      ${lib.optionalString (lprintConfig.papplOptions.contact != null) "Contact ${lprintConfig.papplOptions.contact}"}
+      ${lib.optionalString (lprintConfig.papplOptions.defaultPrinterID != null) "DefaultPrinterID ${lprintConfig.papplOptions.defaultPrinterID}"}
+      ${lib.optionalString
+        (lprintConfig.papplOptions.maxImageSize != null)
+        "MaxImageSize ${
+          if lprintConfig.papplOptions.maxImageSize.maxSize != null then
+            toString lprintConfig.papplOptions.maxImageSize.maxSize
+          else
+            "0"
+        } ${
+          if lprintConfig.papplOptions.maxImageSize.width != null then
+            toString lprintConfig.papplOptions.maxImageSize.width
+          else
+            "0"
+        } ${
+          if lprintConfig.papplOptions.maxImageSize.height != null then
+            toString lprintConfig.papplOptions.maxImageSize.height
+          else
+            "0"
+        }"
+      }
+      UUID ${magicUUID}
+    '' + (lib.concatStringsSep "\n" (lib.attrValues (lib.mapAttrsToList (printerId: (printerConfig: {
+      name = printerId;
+      config = ''
+        <Printer did="" driver="${printerConfig.printerOptions.driver}" id="${printerId}" name="${printerConfig.printerOptions.name}" state="3" uri="${printerConfig.printerOptions.uri}">
+        ${lib.optionalString (printerConfig.papplOptions.dnssdName != null) "DNSSDName ${printerConfig.papplOptions.dnssdName}"}
+        ${lib.optionalString (printerConfig.papplOptions.contact != null) "Contact ${printerConfig.papplOptions.contact}"}
+        ${lib.optionalString
+          (printerConfig.papplOptions.maxImageSize != null)
+          "MaxImageSize ${
+            if printerConfig.papplOptions.maxImageSize.maxSize != null then
+              toString printerConfig.papplOptions.maxImageSize.maxSize
+            else
+              "0"
+          } ${
+            if printerConfig.papplOptions.maxImageSize.width != null then
+              toString printerConfig.papplOptions.maxImageSize.width
+            else
+              "0"
+          } ${
+            if printerConfig.papplOptions.maxImageSize.height != null then
+              toString printerConfig.papplOptions.maxImageSize.height
+            else
+              "0"
+          }"
+        }
+        ${
+          if printerConfig.papplOptions.maxActiveJobs != null then
+            "MaxActiveJobs ${toString printerConfig.papplOptions.maxActiveJobs}"
+          else
+            "MaxActiveJobs 0"
+        }
+        ${
+          if printerConfig.papplOptions.maxCompletedJobs != null then
+            "MaxCompletedJobs ${toString printerConfig.papplOptions.maxCompletedJobs}"
+          else
+            "MaxCompletedJobs 0"
+        }
```

Comment:
don't have config options in the conditional string if it's the same in both sides of the `if else`

don't have the default be null and the type be nullable when if it's nullable you're setting the value to 0, instead set the default to "0" and remove the null check and the `nullOr`